src/states_screens/dialogs/ghost_replay_info_dialog.hpp

```diff
@@ -33,21 +31,31 @@ class GhostReplayInfoDialog : public GUIEngine::ModalDialog
 
 private:
 
-    bool m_self_destroy;
+    bool  m_self_destroy;
 
-    bool m_record_race;
-    bool m_watch_only;
+    bool  m_record_race;
+    bool  m_watch_only;
+    bool* m_compare_ghost;
+
+    unsigned int m_replay_id; // May be updated on list refreshes
+
+    unsigned long long int* m_compare_replay_uid;
```

Comment:
Why is this a pointer?

Reply:
I need to pass back the value to the main state screen, otherwise the value would be lost as soon as the modal dialog is closed.

This works as follow : 
1)GhostReplaySelection passes pointers to two values it needs the dialog to update (one bool to know if there is a comparison active, one to store the UID)
2)The dialog store the pointers value in members to be able to reuse them throughout the class without a complicated scheme to pass them as arguments
3)It uses this to update the values when appropriate
4)When the dialog closes, GhostReplaySelection stlll has the values it needs.

Reply:
GhostReplaySelection::getInstance()->setComapreReplayUid(uint64_t uid);

Reply:
Ok, thanks for the tip, I will try and change it accordingly